doc/ref_arch/openstack/chapters/chapter04.md

```diff
@@ -1,82 +1,209 @@
+-	Large data center capable of hosting thousands of servers and the networking to support them
+-	Mini data center (such as a central office) capable of hosting up to a hundred servers
+-	Edge (not customer premise) capable of hosting between ten to fifty servers
+
+Host profiles (SW Host profile + HW host profile) “partition” the cloud into pseudo sub-clouds, for example, hosts targeted for basic instance types, network intensive instance types and compute intensive instance types. This can happen, because of specific hardware adds and/or hardware and software configurations.  Depending upon the workload types and the capacity requirements, cloud providers and operators may choose to support the instance types with targeted hardware (different number of sockets, RAM, clock speeds, etc.) and host profiles or choose common hardware and minimize the number of host profiles (for example, network intensive and compute intensive types using a common host profile).
+
+As we get away from the large data centers to the smaller sites it becomes progressively difficult to be able to create enough capacity for each of these instance types in support of their target VNFs or to have a mix of hardware targeted for each instance type. 
+
+<a name="4.3.3"></a>
+### 4.3.3. Containerised OpenStack Services
+
+#### 4.3.3.1. Architectural Drivers – Requirements Traceability
+
+| Ref # | sub-category | Description |
+|----|----|-----|
+| req.gen.cnt.02 | Cloud nativeness | The Architecture should consist of service components implemented as microservices that are individually dynamically scalable. |
+| req.vim.02 | General | The Architecture should support deployment of OpenStack components in containers. |
+| req.gen.rsl.01 | Resiliency | The Architecture must support resilient OpenStack components that are required for the continued availability of running workloads. |
+
+#### 4.3.3.2. Justification
+
+ontainers are lightweight compared to Virtual Machines and leads to efficient resource utilization. Kubernetes auto manages scaling, recovery from failures, etc.  Thus, it is recommended that the OpenStack services be containerized for resiliency and resource efficiency.
 
 
 <a name="4.4"></a>
-## 4.4 Cloud Controller Services
-- Core Components (keystone, heat, nova, neutron, glance)
-- Identity Management
+## 4.4 Compute Node Services
 
+In Section 3.5 the high-level set of infrastructure components needed to run VMs was presented.  This section is primarily about scheduling the VM to be created onto a particular physical server. This section delves into the compute host selection for these VMs, the resource pools, hardware and software considerations.
 
-<a name="4.5"></a>
-## 4.5 Cloud Workload Services.
+<p align="center"><img src="../figures/Figure_4_3_Select_host_instance_launch.png" alt="Selection of a compute host to launch an instance"></br>Figure 4-3. Selection of a compute host to launch an instance.</p>
```

Comment:
I think this figure is irrelevant here. 
This chapter should describe what is running on a compute host and not how it is selected.

Reply:
Added to #339